### DISCUSSION TITLE
InSchema vs WithSchema : Why is there inconsistent prepositions?

### DISCUSSION THREAD

**Discussion by KiLLeRRaT** [fluentmigrator/fluentmigrator#1422] [Q&A] [answered]
**Describe the question**
In most cases, when using Fluent Syntax you will use `InSchema(...)` but when you say, create a UniqueConstraint, you have to use `WithSchema(...)`.

e.g.:
```
Alter.Table("ObjectService").InSchema("SCH")...
```
vs

```
Create.UniqueConstraint("UC_Object_ObjectID_ServiceType").OnTable("ObjectService").WithSchema("SCH");
```

Why is that? Why not have it `InSchema` across the board?

Sorry if this is a silly question, but I couldn't find a clear answer anywhere.

Cheers,

EDIT: Just found this inconsistency while writing the `Down()`

```
Create.UniqueConstraint("UC_Object_ObjectID_ServiceType").OnTable("ObjectService").**WithSchema**("SCH");
```
vs
```
Delete.UniqueConstraint("UC_Object_ObjectID_ServiceType").FromTable("ObjectService").**InSchema**("SCH");
```

**Comment by jzabroski** [comment, marked as answer]
I think a lot of these small inconsistencies are due to being an open source project and perhaps less than perfect review process. 

`Create.UniqueConstraint` is another one with some inconsistencies. Its `.Column` and `.Columns` as compared with `Create.Index` using `OnColumn`.

I think the better API might be to remove Preposotions altogether. Get rid of: With, On, From, In, To. Replace with just nouns. For Rename...To we would just do Rename...NewName

I created #1206 to discuss possibly surfacing a different top-level API for FluentMigrator, and allowing you to "import" via `using` statement an alternative syntax from the legacy/current one.

That said, the annoying aspect to me is we historical couldn't guide Intellisense autocompletion with builder pattern hints. I always felt the **Intelli**gence in **Intelli**sense was not sophisticated.  However, with Roslyn C# Completion Service being open sourced, it looks like [it may be possible](https://www.strathweb.com/2018/12/using-roslyn-c-completion-service-programmatically/).

**Reply by jzabroski** [reply]
@KiLLeRRaT I moved this Issue to a Discussion and marked the above as answer, as well as updated my reply.  Let me know if you still have any related questions or thoughts you want to discuss.

**Reply by KiLLeRRaT** [reply]
Hi @jzabroski , No thanks :) I was just curious about it all and if there was a reason, but it appears that it was just some inconsistincies, and there's no reason.  It's easy enough to find the correct method to use, so probably not worth _fixing_ as such.

Cheers!
